assignment/impl/src/java/org/sakaiproject/assignment/impl/AssignmentServiceImpl.java

```diff
@@ -1633,6 +1632,7 @@ private String getFormattedStatus(AssignmentConstants.SubmissionStatus subStatus
         }
         if (submitTime != null && submission.getReturned() && returnTime != null && returnTime.isBefore(submitTime)) {
```

Comment:
I know this wasn't part of your commit, but it seems to me the variable submitTime should be renamed to latestSubmitTime, to make it clearer. "returnTime.isBefore(latestSubmitTime)" makes way more sense.